### PR TITLE
fix: Updated the logic to calculate estimated effort to complete a node in learner pathway.

### PR DESCRIPTION
**Pull request by saleem-latif**
__JIRA Ticket:__ [ENT-5646](https://openedx.atlassian.net/browse/ENT-5646)

__Description:__
Updated the model so that the step reflects the min/max range of values.

### CODE REVIEW

**Review by sameenfatima78 — APPROVED** [review]

File: course_discovery/apps/learner_pathway/models.py [hunk 1]
```diff
@@ -181,8 +181,21 @@ def remove_node(self, uuid):
...
+            (tuple<int, int>): A tuple containing the min and max of estimated time of completion.
+        """
+        estimated_completion_times_of_nodes = sorted(
+            [node.get_estimated_time_of_completion() for node in self.get_nodes()]
+        )
+        estimated_completion_times_of_nodes = sorted(estimated_completion_times_of_nodes)
+        return (
+            sum(estimated_completion_times_of_nodes[:self.min_requirement]),
+            sum(estimated_completion_times_of_nodes[-self.min_requirement:]),
+        )
```
Comment (muhammad-ammar) [inline course_discovery/apps/learner_pathway/models.py]: Can we please add some explanation regarding how the above code works?

**Review by muhammad-ammar — APPROVED** [review]
LGTM. Please see my comment regarding improving code documentation.

File: course_discovery/apps/learner_pathway/models.py [hunk 2]
```diff
@@ -181,8 +181,21 @@ def remove_node(self, uuid):
...
+        """
+        Get a range of estimated time of completion.
+
+        Returns:
+            (tuple<int, int>): A tuple containing the min and max of estimated time of completion.
+        """
+        estimated_completion_times_of_nodes = sorted(
+            [node.get_estimated_time_of_completion() for node in self.get_nodes()]
+        )
+        estimated_completion_times_of_nodes = sorted(estimated_completion_times_of_nodes)
```
Comment (muneebGH) [inline course_discovery/apps/learner_pathway/models.py]: we are sorting the list two times(one in upper line) is this ok?

File: course_discovery/apps/learner_pathway/models.py [hunk 3]
```diff
@@ -123,7 +123,7 @@ def time_of_completion(self) -> float:
         """
         completion_time = 0
         for step in self.steps.all():
-            completion_time += step.get_estimated_time_of_completion()
+            completion_time += avg(step.get_estimated_time_of_completion())
```
Comment (muneebGH) [inline course_discovery/apps/learner_pathway/models.py:126]: if there is a block in the step and that returns 'Not known' through [this](https://github.com/openedx/course-discovery/blob/850b8b97a22ef94bec08121bfbd76c92620ead0a/course_discovery/apps/learner_pathway/models.py#L321)
that string will be a part of list and an exception might be thrown, what do you? plus do we need a separate function for avg? can we like paste the one liner here?

File: course_discovery/apps/learner_pathway/models.py (hunk 1 above)
Comment (saleem-latif) [reply course_discovery/apps/learner_pathway/models.py]: Done

File: course_discovery/apps/learner_pathway/models.py (hunk 2 above)
Comment (saleem-latif) [reply course_discovery/apps/learner_pathway/models.py]: I forgot to remove this line, good catch

File: course_discovery/apps/learner_pathway/models.py (hunk 3 above)
Comment (saleem-latif) [reply course_discovery/apps/learner_pathway/models.py:126]: I have updated the return type of block to `int` all `get_estimated_time_of_completion` methods should have the same type as the base class `LearnerPathwayNode`. 
Regarding `avg` it seemed more readable to add a separate. function for this.

**Review by muneebGH — APPROVED** [review]
LGTM

File: course_discovery/apps/learner_pathway/models.py (hunk 3 above)
Comment (muneebGH) [reply course_discovery/apps/learner_pathway/models.py:126]: oky cool

**Pull request merged**